Compiler/src/abstractinterpretation.jl

```diff
@@ -3560,7 +3560,7 @@ function global_assignment_rt_exct(interp::AbstractInterpreter, sv::AbsIntState,
             Pair{Any,Any}(newty, Union{ErrorException, TypeError}), nothing)
     end
     partition = abstract_eval_binding_partition!(interp, g, sv)
-    return Pair{Pair{Any,Any}, Union{Core.BindingPartition, Nothing}}(
+    return Pair{Pair{Any,Any}, Core.BindingPartition}(
```

Comment:
This is to keep the rt consistent with the other path in this function

Reply:
Ok, I will revert this one.